modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -109,20 +109,27 @@ export default class Tile3DLayer extends CompositeLayer {
   }
 
   async _loadTileset(tilesetUrl) {
-    const {loadOptions} = this.props;
+    const {loadOptions = {}} = this.props;
 
     // TODO: deprecate `loader` in v9.0
     let loader = this.props.loader || this.props.loaders;
     if (Array.isArray(loader)) {
       loader = loader[0];
     }
 
-    const options = {...loadOptions};
+    const options = {loadOptions: {...loadOptions}};
     if (loader.preload) {
       const preloadOptions = await loader.preload(tilesetUrl, loadOptions);
+
+      if (preloadOptions.headers) {
+        options.loadOptions.fetch = {
+          ...options.loadOptions.fetch,
+          headers: preloadOptions.headers
```

Comment:
If `preloadOptions` is supposed to be a top-level loaders.gl options object, then `preloadOptions.headers` should be moved to `preloadOptions.fetch.headers` on the loaders side?

Reply:
Yeah. It's already done and Cesium tilesets don't work on loaders.gl now.

Reply:
I'm saying the `preload` function should return an object that contains `fetch`, in compliance with the latest options format.

Reply:
Well. I see here
`preloadOptions.headers` -> `options.loadOptions.fetch.headers`.
Anyway it works. I checked with `yarn start-local-deck`

Reply:
I know it works, but this change should not be handled by deck.gl. It's not deck's responsibility to fix the format of something that is originated from loaders.gl in order to feed it back to loaders.gl.

Reply:
This is data that Cesium returns. @ibgreen what do you think if we add some logic in `preload` function?